pkg/cluster/internal/create/actions/installcni/cni.go

```diff
@@ -83,11 +86,39 @@ func (a *action) Execute(ctx *actions.ActionContext) error {
 		manifest = out.String()
 	}
 
+	// Add the controlplane endpoint so kindnet doesn´t have to wait for kube-proxy
+	controlPlaneEndpoint, err := ctx.Provider.GetAPIServerInternalEndpoint(ctx.Config.Name)
+	if err != nil {
+		return err
+	}
+
+	patchValue := fmt.Sprintf(`
+- op: add
+  path: /spec/template/spec/containers/0/env/-
+  value:
+    name: CONTROL_PLANE_ENDPOINT
+    value: %s
+`, controlPlaneEndpoint)
+
+	controlPlanePatch6902 := config.PatchJSON6902{
+		Group:   "apps",
+		Version: "v1",
+		Kind:    "DaemonSet",
+		Patch:   patchValue,
+	}
+
+	patchedConfig, err := patch.KubeYAML(manifest, nil, []config.PatchJSON6902{controlPlanePatch6902})
```

Comment:
TODO: only patch this if it is kindnetd / wants to be patched (similar to the approach with templating previously with the sentinel comment in the yaml)